### PR TITLE
feat: using Python 3.13

### LINKED ISSUE #537: [MAINTENANCE] Support for Python 3.13
**DISCLAIMER**: This issue was created by a bot in behalf of the PyAnsys Core team. We have not verified whether your repository is already supporting Python 3.13. If that's the case, please close this issue and let us know by commenting in it.

## Description

According to our internal guidance, we should start active support for Python 3.13 across all repositories in the PyAnsys ecosystem. Up until now, most of our projects have been blocked by some dependencies that were not compatible with Python 3.13. However, most of these dependencies have been updated to support Python 3.13, and we are now able to move forward with this change.

See [ADR-21](https://adrs.ansys.com/docs/adrs/0021-python-version-support/) for more information. Specifically, the section corresponding to [Adding support](https://adrs.ansys.com/docs/adrs/0021-python-version-support/#adding-support-2)

## Action Items

1. Build the wheelhouse of the various repositories using Python 3.13.
2. Build the documentation using Python 3.13 (whenever possible).
3. Run the test suites using Python 3.13 (whenever possible).

Steps 2 and 3 may not be possible or undesired from your side, in case you want to stick to a different version of Python for the documentation and testing. Most repositories have a ``MAIN_PYTHON_VERSION`` environment variable that we are suggesting to bump to 3.13.

## Reference Pull Request

For your reference, you can check the following PR that was created to bump the Python version to 3.13 in the ``pyansys-tools-versioning`` repository.

Example PR: [PyAnsys Tools Versioning](https://github.com/ansys/pyansys-tools-versioning/pull/316)

## Additional Information

Please ensure that all necessary updates and modifications are made to support Python 3.13 effectively. If there are any issues or concerns, feel free to discuss them in this issue.

Thank you for your cooperation.

### Extra: blocking dependencies for Python 3.13 and their status

Most of the PyAnsys projects have a dependency on the ``vtk`` package, coming from its usage in the ``pyvista`` package. The ``vtk`` package has been updated to version ``9.4.0`` which now supports Python 3.13.

PyVista has decided to block its usage on version 0.44.2, while they work on supporting it for 0.43.0. This is due to the fact that there are some incompatibilities being addressed on their side. However, the impact on our libraries depends on whether we are making use of those advanced, impacted features. If we are not using them, we can start supporting Python 3.13 (while using a different PyVista version other than 0.44.2).

The list of incompatibilities are being tracked in the following issue ``https://github.com/pyvista/pyvista/issues/6731``. Feel free to check it out.

We also recommend that the projects using self-hosted runners, both Windows and Linux based, upgrade the ``pyvista/setup-headless-display-action`` from version ``v2`` to ``v3``. This is due to the fact that VTK will require new Mesa3D library versions for running GPU-less rendering. This is not a blocker for the Python 3.13 support, but it is a good practice to keep the runners up-to-date.

---

If you want this repository to be excluded from this automated maintenance process, please let us know by [filling in the opt-out request form](https://github.com/ansys-internal/pyansys-maintenance-automation/issues/new?labels=&projects=&template=opt-out.yml&title=Request+to+opt-out+from+automated+maintenance+for+ansys%2Fpyansys-dev-guide).

### PR DESCRIPTION
**Pull request by RobPasMue**
As title says. Closes #537

### CODE REVIEW

**Comment by RobPasMue** [comment]
Also took advantage to upgrade action versions

**Comment by RobPasMue** [comment]
Done everything on the web browser so... checkout inconsistencies

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
       - main
 
 env:
-  MAIN_PYTHON_VERSION: '3.12'
+  MAIN_PYTHON_VERSION: '3.13'
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:12]:
```suggestion
  MAIN_PYTHON_VERSION: '3.12'
```

File: .github/workflows/nightly-build.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ on:
   workflow_dispatch:
 
 env:
-  MAIN_PYTHON_VERSION: '3.10'
+  MAIN_PYTHON_VERSION: '3.13'
```
Comment (RobPasMue) [inline .github/workflows/nightly-build.yml:9]:
```suggestion
  MAIN_PYTHON_VERSION: '3.12'
```

**Review by RobPasMue — COMMENTED** [review]
Let's try with 3.12 as main version... this xvfb.run issues are driving me nuts

**Comment by RobPasMue** [comment]
Nah... it didn't fix anything. No clue about what's going on

**Comment by MaxJPRey** [comment]
> Nah... it didn't fix anything. No clue about what's going on

I will try to have a look.

**Comment by moe-ad** [comment]
> Nah... it didn't fix anything. No clue about what's going on

@RobPasMue @MaxJPRey,  if you do figure out the root cause of the xvfb issue, please share. It is also holding us up on ansys/pyansys-dev-guide#536.

**Comment by RobPasMue** [comment]
Whelp, using the ``pyvista/setup-headless-display-action`` solved it! Merging :)

**Pull request merged**